### PR TITLE
Solutions to Issues #31, #48 and Additional Code Refactoring

### PR DESCRIPTION
**Pull request by mreyeros**
- Introduced Blazored.FluentValidation for handling Validation of ShortUrlEntity on Edits or Adds
- Introduced IStorageTableHelper to inject StorageTableHelper via Dependency Injection for all Azure Functions removing code duplication
- Modified ShortUrlEntity Clicks definition from int to Dictionary<string, int> to store click counts by country
- Intriduced some helpers and extensions methods to reduce code duplication
- Introduced EntityJsonPropertyConverter to handle serialization/deserialization of ShortUrlEntity Clicks property
- Introduced UserIpLocationService to call the https://freegeoip.app/ service for obtaining user click data (i.e. country of origin)
- Updated all azure functions to inject IStorageTableHelper to reduce code duplication and coupling to StorageTableHelper implementation

### CODE REVIEW

**Comment by FBoucher** [comment]
Wow very nice work @mreyeros !!!

Let's plan some session to test all those super cool new features.

**Comment by mreyeros** [comment]
:) You are welcome Frank, absolutely lets do it!

File: src/shortenerTools/Domain/StorageTableHelper.cs [hunk 1]
```diff
@@ -1,50 +1,54 @@
...
         }
 
-        private  CloudTable GetTable(string tableName){
-            CloudStorageAccount storageAccount = this.CreateStorageAccountFromConnectionString();
-            CloudTableClient tableClient = storageAccount.CreateCloudTableClient(new TableClientConfiguration());
-            CloudTable table = tableClient.GetTableReference(tableName);
+        private CloudTable GetTable(string tableName)
+        {
+            var storageAccount = this.CreateStorageAccountFromConnectionString();
+            var tableClient = storageAccount.CreateCloudTableClient(new TableClientConfiguration());
```
Comment (DavidTCarpenters) [inline src/shortenerTools/Domain/StorageTableHelper.cs:39]: Out of scope for this PR, but the guidance is to use a static table client

File: src/adminTools/adminBlazorWebsite/src/Data/UrlShortenerService.cs [hunk 2]
```diff
@@ -47,17 +47,13 @@ public async Task<ShortUrlList> GetUrlList()
...
-            {
-                using (var response = await client
-                    .SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken)
-                    .ConfigureAwait(false))
-                {
-                    var resultList = response.Content.ReadAsStringAsync().Result;
-                    return JsonConvert.DeserializeObject<ShortUrlList>(resultList);
-                }
-            }
+            using var client = new HttpClient();
```
Comment (DavidTCarpenters) [inline src/adminTools/adminBlazorWebsite/src/Data/UrlShortenerService.cs:50]: Can this be injected, creating http clients inline will to lead to exhaustion

**Comment by DavidTCarpenters** [comment]
Looking good

**Comment by DavidTCarpenters** [comment]
Needs the lastest from branch vnext merging into and conflicts resolved

**Comment by FBoucher** [comment]
FYI @DavidTCarpenters I started the v3 branch to use the Identity of SWA and really simplify the deployment.

**Pull request closed without merging**